core/src/main/java/org/apache/iceberg/util/ParallelIterable.java

```diff
@@ -126,6 +141,7 @@ private boolean checkTasks() {
             } catch (InterruptedException e) {
               throw new RuntimeException("Interrupted while running parallel task", e);
             }
+            continuation.ifPresent(yieldedTasks::addLast);
```

Comment:
Why call this outside of the `try` block? That block only handles `ExecutionException` and `InterruptedException` so it seems safe to call this inside and not need to separate the logic into 3 places.

Reply:
i generally prefer not to rely on specific types in catch clauses, since it's easy to overlook something (especially as the code evolves).  will move nonetheless